packages/sparrowql/src/build.ts

```diff
@@ -10,321 +11,322 @@ export type Relation = {
+
+    steps.push({
+      relation: Object.assign({}, relation, {
+        local: relative([relation.from, relation.local].join('.'), false),
+        toAlias: aliases[relation.to] || relation.to
+      })
+    });
+
+    inlineMatch();
+    inlineSort();
+  }
+
+  function inlineSort() {
+    if (
+      !sort ||
+      needed.some(
+        collection =>
+          !joined.includes(collection) &&
+          isComputed(collection) &&
+          !computed[stripComputed(collection)].mapper
+      )
+    )
+      return;
+
+    const entries = Object.entries(sort);
+    if (entries.length === 0) {
+      sort = undefined;
+      return;
     }
 
-    function inlineSort() {
-        if (
-            !sort ||
-            needed.some(
-                collection =>
-                    !joined.includes(collection) &&
-                    isComputed(collection) &&
-                    !computed[stripComputed(collection)].mapper
-            )
+    if (entries.every(entry => joined.includes(getNameCollection(entry[0])))) {
+      steps.push({
+        sort: entries.reduce(
+          (object, entry) => Object.assign(object, {[relative(entry[0], false)]: entry[1]}),
+          {}
         )
-            return;
-
-        const entries = Object.entries(sort);
-        if (entries.length === 0) {
-            sort = undefined;
-            return;
-        }
-
-        if (entries.every(entry => joined.includes(getNameCollection(entry[0])))) {
-            steps.push({
-                sort: entries.reduce(
-                    (object, entry) => Object.assign(object, {[relative(entry[0], false)]: entry[1]}),
-                    {}
-                )
-            });
-            sort = undefined;
-        }
+      });
+      sort = undefined;
     }
+  }
 }
 
-const translateOperators = {
-    group: (start: string, step: Step) => [{$group: step.group}],
-    limit: (start: string, step: Step) => [{$limit: step.limit}],
-    match: (start: string, step: Step) => [{$match: step.match}],
-    projection: (start: string, step: Step) => [{$project: step.projection}],
-    relation: (start: string, step: Step) => [
-        {
-            $lookup: {
-                as: getNameRelative(start, step.relation!.to, false),
-                foreignField: step.relation!.foreign,
-                from: step.relation!.toAlias,
-                localField: step.relation!.local
-            }
-        },
-        {
-            $unwind: {
-                path: getNameRelative(start, step.relation!.to, true),
-                preserveNullAndEmptyArrays: true
-            }
-        }
-    ],
-    skip: (start: string, step: Step) => [{$skip: step.skip}],
-    sort: (start: string, step: Step) => [{$sort: step.sort}]
+const translateOperators: Record<'group' | 'limit' | 'match' | 'projection' | 'relation' | 'skip' | 'sort', <T>(start: string, step: Step<T>) => any> = {
+  group: <T = {}>(start: string, step: Step<T>) => [{$group: step.group}],
```

Comment:
Inline typings here are not needed, as `translateOperators` is completely typed.